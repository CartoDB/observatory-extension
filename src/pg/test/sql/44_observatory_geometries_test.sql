\pset format unaligned
\set ECHO none
SET client_min_messages TO WARNING;

-- set up variables for use in testing

\set cartodb_census_tract_geometry '0106000020E6100000010000000103000000010000001600000054EF703B347C52C053FF2092215B44400E8FC4CBD37B52C0B86C74CE4F5B4440C743DFDDCA7B52C05BDDB1D8265B44402A4A09C1AA7B52C06963EC84975A44403B873254C57B52C02CAC1BEF8E5A44401A9B1DA9BE7B52C03E6F2A52615A4440DAC2F352B17B52C06F3FE081015A444040446ADAC57B52C0EEDFBC38F159444033BC5983F77B52C01F5ED72FD8594440CE2D7425027C52C0A66DC328085A44400EECF82F107C52C0866D8B321B5A444098F221A81A7C52C0CFEFFB372F5A444049ED45B41D7C52C0775DBF60375A4440C458A65F227C52C0D70DDB16655A4440F82EA52E197C52C09AA73AE4665A4440DDE522BE137C52C00464AF777F5A44405498BED7107C52C04659BF99985A444011D90759167C52C09330D3F6AF5A444043679945287C52C01E680586AC5A4440FE59F3E32F7C52C0C9290131095B4440F76871C6307C52C09405137F145B444054EF703B347C52C053FF2092215B4440'

\set cartodb_county_geometry '0106000020E610000001000000010300000001000000BC00000026874F3A918352C0A36B26DF6C534440FDB627486C8352C0DB9E20B1DD534440B57F65A5498352C0F53DEAAF5754444010FC6F253B8352C0411B800D885444401B739EB12F8352C091DFA293A554444003A73E90BC8252C0DA4C857824564440BB1FB75F3E8252C08E368E588B574440C414AC71368252C0371268B0A9574440CDA44DD53D8152C04E1F813FFC564440B2EDB435228152C000D9EBDD1F5744401969A9BC1D8152C0783D98141F5744400C4A9869FB8052C0C81EA1664857444019F8510DFB8052C0C41A2E724F5744409E4FE449D28052C02820ED7F80574440B33C0FEECE8052C0D7614CFA7B5744403380B740828052C0A23CF372D8574440BE9A0304738052C045ED7E15E05744406279573D608052C0A3A487A1D55744401778EE3D5C8052C03FB5C189E8574440963270404B8052C0ECCEDA6D1758444085C43D963E8052C0DB9DB5DB2E5844407B478D09318052C03FA374E95F58444054336B29208052C067E55E6056584440A92688BA0F8052C026F6D03E56584440F288D1730B8052C0D11C59F965584440C5A69542208052C0B67BB94F8E584440C06856B60F8052C06D15C440D7584440387CD289048052C0C987A06AF4584440F8BD4D7FF67F52C0DA9A745B22594440F1CCCB61F77F52C0047FBF982D5944404AADF71BED7F52C0AFD991EA3B59444021179CC1DF7F52C000D8800871594440447F68E6C97F52C060C26856B65944406F404B57B07F52C0FADEA63FFB594440A3E2FF8EA87F52C00BFFE9060A5A44402A4A09C1AA7F52C0489F56D11F5A44404201DBC1887F52C0C9F10A444F5A4440D5A88768747F52C0B61E85EB515A44407B0C569C6A7F52C0F5CFD380415A444031E202D0287F52C012B534B7425A4440646A12BC217F52C015A243E0485A4440D1E0B6B6F07E52C0FECA4A93525A444045B3B27DC87E52C06EB9FAB1495A4440B75A272EC77E52C03B821B295B5A44409907B0C8AF7E52C07ED93D79585A44401A82E3326E7E52C03D170D198F5A4440893C49BA667E52C0D519DF17975A4440243ACB2C427E52C03D40F7E5CC5A4440CF26874F3A7E52C060C5A9D6C25A4440152EABB0197E52C006B5DFDA895A4440B031AF230E7E52C0D1C43BC0935A4440194CC3F0117E52C0EFFACC599F5A444029836A83137E52C0E35C8AABCA5A4440C8737D1F0E7E52C0B276DB85E65A4440148C4AEA047E52C08C7E349C325B4440258C6665FB7D52C0528A1D8D435B44405B1B2AC6F97D52C064187783685B44400F9546CCEC7D52C07DDB1324B65B44401B261AA4E07D52C0B9641C23D95B444072672618CE7D52C0990645F3005C44401A2AC6F99B7D52C01B2444F9825C4440A6ACA6EB897D52C0FC2D01F8A75C4440AC4F39268B7D52C0A992C83EC85C44408F847D3B897D52C02C5393E00D5D4440D8CC21A9857D52C0631F6459305D4440ECED96E4807D52C05F048D99445D44404EAF9465887D52C03DC91D36915D4440BC38F1D58E7D52C0CDF753E3A55D444053556820967D52C0B2226AA2CF5D4440BDD7101C977D52C0C7B4368DED5D4440D37D00529B7D52C0EAE1CB44115E44409731056B9C7D52C08FA2CEDC435E4440A447533D997D52C04DD026874F5E4440D2139678407D52C024A435069D5E44406B753925207D52C0C954C1A8A45E4440391BF2CF0C7D52C0DF7CEA58A55E44406F08556AF67C52C0CB24EA059F5E444002CE52B29C7C52C06A239EEC665E444013200A664C7C52C07E315BB22A5E4440C3758C2B2E7C52C09B4B7155D95D4440ED08A7052F7C52C0434B1E4FCB5D4440FF56B263237C52C07C39B35DA15D4440AB8BDB68007C52C0AA90F2936A5D444024E4839ECD7B52C0385E81E8495D444017265305A37B52C0C1DCEEE53E5D44402581CD39787B52C0675721E5275D44408C0C7217617B52C0BE5E61C1FD5C44402F630AD6387B52C081C64CA25E5C444080E7DEC3257B52C0A23B889D295C4440629AE95E277B52C0D27FD6FCF85B4440322A70B20D7B52C0D726158DB55B444098EFE0270E7B52C06562BA10AB5B4440D2F36E2C287B52C0340873BB975B4440870CAB78237B52C08A8A389D645B4440AE928FDD057B52C05069C4CC3E5B44407D8978EBFC7A52C0FB1BB4571F5B44409BF0129CFA7A52C05B86FF74035B444026BD6F7CED7A52C0D669A4A5F25A4440699C4D47007B52C0E2F4F57CCD5A4440E250BF0B5B7A52C04DEE77280A5A4440CD18E6046D7A52C000FA7DFFE6594440DA899290487A52C0EC7E15E0BB594440CA0EF10F5B7A52C0229D819197594440A15EF0694E7A52C0B240BB438A5944402DC55565DF7952C0E0E995B20C594440DF2D90A0F87952C0729EB12FD9584440B1666490BB7952C0ADEE586C935844401D34BBEEAD7952C0169B560A81584440A197512CB77952C0269831056B5844400CE02D90A07952C0B2D0CE6916584440779961A3AC7952C0A5D1E4620C58444075FEEDB25F7952C014713AC9565744404FD7DAFB547952C0544FE61F7D574440A579C7293A7952C09BC0745AB757444060FBC9181F7952C096A608707A57444074AE2825047952C054DAE21A9F57444073779D0DF97852C0B37405DB885744408B48145AD67852C08F12BBB6B757444062F030ED9B7852C08EBA9D7DE55744403EFD67CD8F7852C06F9010E50B5844405691D101497852C09C7DE5417A58444018C1C6F5EF7752C0B7162D40DB5844400E7C0C569C7752C03DE8D9ACFA584440A561F888987752C02F2AA913D05844408B63247B847752C0B48993FB1D584440DA317557767752C0057767EDB65744400C6C9560717752C052FEB5BC72574440FD9C82FC6C7752C0DF46CA164957444036DC476E4D7752C031569BFF575744406D4E2503407752C05B4C33DDEB56444087F546AD307752C03DCBF3E0EE564440A335069D107752C0C6E88024EC55444019BE8575E37652C0C014E5D2F8554440829ECDAACF7652C0262D5C56615544405FAB048BC37652C060F20698F9544440EFF96184F07652C06EB72407EC544440C663062AE37652C0EE3B86C77E5444401E4AB4E4F17652C0DFD407927754444056B26323107752C0F577B6476F544440C1F693313E7752C0BE5AB56B42544440C3758C2B2E7752C03C0801F91254444021C2BF081A7752C0237F30F0DC534440A2E4D539067752C064170CAEB9534440C0FA3F87F97652C058F5B9DA8A5344408AFA2477D87652C081A62556465344404922FB20CB7652C0755089EB18534440E286DF4DB77652C01E63EE5A425244400E7D772B4B7652C004A568E55E524440735B22179C7552C0E0CCAFE600514440086B63EC847552C0A992C83EC850444088CE328B507552C0152D40DB6A504440C7ED974F567552C05546239F5750444013713AC9567552C057DC7F643A50444044460724617552C0A84885B1854E4440F52686E4647552C02B978DCEF94D444035AE7FD7677552C016AE0E80B84D4440B8E00CFE7E7552C0AE1F6283854D4440436B459BE37552C0C3E23785954C444011A27C410B7652C03B1405FA444C4440568B4F01307652C0AF58C345EE4B4440A5457D923B7652C07A28D192C74B44402B9CA4F9637652C0A1957B81594B4440A61ABD1AA07652C02F0C5872154B4440110CE71A667752C0C4A2E9EC644A4440BE28417FA17752C07E99EFE0274A444052C90050C57752C0B04B546F0D4A44401DFB592C457852C09D39EB538E494440046D72F8A47852C06B8E739B70494440F39F353FFE7852C0B8A0BE654E4944406033C005D97952C0D762B7CF2A494440DF783E03EA7952C04BDA54DD234944408399B67F657B52C035902E36AD48444080CEA44DD57B52C0421B800D88484440B4A50EF27A7C52C07680272D5C4844405F4D6551D87C52C05BFDD8243F48444064A54929E87C52C0C35E28603B48444023AB22DC647D52C03C0801F9124844401111FE45D07F52C0F0CEA10C55474440B704E09F528252C0775C548B88464440A9B4C5353E8252C0BCC117265349444031CB9E04368252C0F5285C8FC24944404F11A7936C8252C09EE238F06A4B44402152D32EA68252C0BB02D193324D444067075C57CC8252C0EBDCB419A74D4440880629780A8352C015DC0F78604E444054F833BC598352C09E91088D604F44403B7E6FD39F8352C05B1ABFF04A50444026874F3A918352C0A36B26DF6C534440'

-- OBS_GetBoundary tests

-- expect most recent census tract boundary at cartodb nyc
-- timespan implictly null
SELECT st_envelope(cdb_observatory.OBS_GetBoundary(
  cdb_observatory._TestPoint(),
  'us.census.tiger.census_tract'
)) = st_envelope(st_asewkb(:'cartodb_census_tract_geometry')) As OBS_GetBoundary_cartodb_census_tract;

-- expect most recent census county boundary (brooklyn) at cartodb nyc
-- timespan implictly null
SELECT st_envelope(cdb_observatory.OBS_GetBoundary(
  cdb_observatory._TestPoint(),
  'us.census.tiger.county'
)) = st_envelope(st_asewkb(:'cartodb_county_geometry')) As OBS_GetBoundary_cartodb_county;

-- expect null geometry since boundary_id is null
-- timespan implictly null
SELECT cdb_observatory.OBS_GetBoundary(
  cdb_observatory._TestPoint(),
  'us.census.tiger.non_existent'
) IS NULL As OBS_GetBoundary_non_existent_boundary_id;

-- expect null geometry since there are no census tracts at null island
-- timespan implictly null
SELECT cdb_observatory.OBS_GetBoundary(
  ST_SetSRID(ST_MakePoint(0, 0), 4326),
  'us.census.tiger.census_tract'
) IS NULL As OBS_GetBoundary_null_island_census_tract;

-- expect census tract boundary at cartodb nyc from 2014
SELECT st_envelope(cdb_observatory.OBS_GetBoundary(
  cdb_observatory._TestPoint(),
  'us.census.tiger.census_tract',
  '2015'
)) = st_envelope(st_asewkb(:'cartodb_census_tract_geometry')) As OBS_GetBoundary_year_census_tract;

-- should return null
-- look for census tracts a year before census released them
SELECT cdb_observatory.OBS_GetBoundary(
  cdb_observatory._TestPoint(),
  'us.census.tiger.census_tract',
  '1988'
) IS NULL As OBS_GetBoundary_unlisted_year;

-- OBS_GetBoundaryId tests

-- should give back '36047048500', the geoid of cartodb's census tract
SELECT cdb_observatory.OBS_GetBoundaryId(
  cdb_observatory._TestPoint(),
  'us.census.tiger.census_tract'
) = '36047048500'::text As OBS_GetBoundaryId_cartodb_census_tract;

-- should give back '36047048500', the geoid of cartodb's census tract
SELECT cdb_observatory.OBS_GetBoundaryId(
  cdb_observatory._TestPoint(),
  'us.census.tiger.census_tract',
  '2015'
) = '36047048500'::text As OBS_GetBoundaryId_cartodb_census_tract_with_year;

-- should give back '36047', the geoid of cartodb's county (King's/
--  Brooklyn, NY)
SELECT cdb_observatory.OBS_GetBoundaryId(
  cdb_observatory._TestPoint(),
  'us.census.tiger.county',
  '2015'
) = '36047'::text As OBS_GetBoundaryId_cartodb_county_with_year;

-- should give back null since there is not a census tract at null island
SELECT cdb_observatory.OBS_GetBoundaryId(
  ST_SetSRID(ST_MakePoint(0, 0), 4326),
  'us.census.tiger.census_tract'
) IS NULL As OBS_GetBoundaryId_null_island;

-- OBS_GetBoundaryById

-- should give geometry of King's County/Brooklyn, NY

SELECT st_envelope(cdb_observatory.OBS_GetBoundaryById(
  '36047',
  'us.census.tiger.county'
)) = st_envelope(st_asewkb(:'cartodb_county_geometry')) As OBS_GetBoundaryById_cartodb_county;

-- Should match output of GetBoundary on similar inputs
SELECT cdb_observatory.OBS_GetBoundaryById(
  '36047', -- cartodb's county
  'us.census.tiger.county'
) = cdb_observatory.OBS_GetBoundary(
  cdb_observatory._TestPoint(), -- CartoDB's office
  'us.census.tiger.county'
) As OBS_GetBoundaryById_compared_with_obs_GetBoundary;

-- should give null since boundary_id does not match geometry reference id
SELECT st_envelope(cdb_observatory.OBS_GetBoundaryById(
  '36047',
  'us.census.tiger.county',
  '2015'
)) = st_envelope(st_asewkb(:'cartodb_county_geometry')) OBS_GetBoundaryById_boundary_id_mismatch_geom_id;

-- should give null since boundary_id does not match geometry reference id
SELECT cdb_observatory.OBS_GetBoundaryById(
  '36047',
  'us.census.tiger.census_tract'
) IS NULL As OBS_GetBoundaryById_boundary_id_mismatch_geom_id;

-- _OBS_GetBoundariesByGeometry
-- check that all census tracts intersecting with the geometry are returned
-- order them to ensure that the same values are returned
SELECT
  array_agg(geom_refs) = Array['36047025700','36047028501','36047038900','36047039100','36047042300','36047042500','36047042700','36047044900','36047045300','36047048500','36047048900','36047049100','36047049300','36047050500','36047050700'] As _OBS_GetBoundariesByGeometry_tracts_around_cartodb
FROM (
  SELECT *
  FROM cdb_observatory._OBS_GetBoundariesByGeometry(
    -- near CartoDB's office
    ST_MakeEnvelope(-73.9452409744,40.6988851644,-73.9280319214,40.7101254524,
                    4326),
    'us.census.tiger.census_tract')
  ORDER BY geom_refs ASC
) As m(the_geom, geom_refs);

-- Null Island area
SELECT
  array_length(array_agg(geom_refs), 1) IS NULL As _OBS_GetBoundariesByGeometry_tracts_around_null_island
FROM (
  SELECT *
  FROM cdb_observatory._OBS_GetBoundariesByGeometry(
    -- around null island
    ST_MakeEnvelope(-0.1400756836,-0.2114863362,0.1455688477,0.2059932086,
                    4326),
    'us.census.tiger.census_tract')
  ORDER BY geom_refs ASC
) As m(the_geom, geom_refs);

-- OBS_GetBoundariesByGeometry

-- check that all census tracts intersecting with the geometry are returned
-- order them to ensure that the same values are returned
SELECT
  array_agg(geom_refs) = Array['36047025700','36047028501','36047038900','36047039100','36047042300','36047042500','36047042700','36047044900','36047045300','36047048500','36047048900','36047049100','36047049300','36047050500','36047050700'] As OBS_GetBoundariesByGeometry_tracts_around_cartodb
FROM (
  SELECT *
  FROM cdb_observatory.OBS_GetBoundariesByGeometry(
    -- near CartoDB's office
    ST_MakeEnvelope(-73.9452409744,40.6988851644,-73.9280319214,40.7101254524,
                    4326),
    'us.census.tiger.census_tract')
  ORDER BY geom_refs ASC
) As m(the_geom, geom_refs);

-- Null Island area
SELECT
  array_length(array_agg(geom_refs), 1) IS NULL As OBS_GetBoundariesByGeometry_tracts_around_null_island
FROM (
  SELECT *
  FROM cdb_observatory.OBS_GetBoundariesByGeometry(
    -- around null island
    ST_MakeEnvelope(-0.1400756836,-0.2114863362,0.1455688477,0.2059932086,
                    4326),
    'us.census.tiger.census_tract')
  ORDER BY geom_refs ASC
) As m(the_geom, geom_refs);

-- OBS_GetBoundariesByPointAndRadius

-- check that all census tracts intersecting with the geometry are returned
-- order them to ensure that the same values are returned
SELECT
  array_agg(geom_refs) = Array['36047038900','36047039100','36047042500','36047042700','36047045300','36047048500','36047048900','36047049100','36047049300'] As OBS_GetBoundariesByPointAndRadius_around_cartodb
FROM (
  SELECT *
  FROM cdb_observatory.OBS_GetBoundariesByPointAndRadius(
    -- 500 meter circle centered on CartoDB's office
    cdb_observatory._testPoint(),
    500,
    'us.census.tiger.census_tract')
  ORDER BY geom_refs ASC
) As m(the_geom, geom_refs);

-- Null Island area
SELECT
  array_length(array_agg(geom_refs), 1) IS NULL As OBS_GetBoundariesByPointAndRadius_around_null_island
FROM (
  SELECT *
  FROM cdb_observatory.OBS_GetBoundariesByPointAndRadius(
    -- around null island
    ST_SetSRID(ST_Point(0, 0), 4326),
    500,
    'us.census.tiger.census_tract')
  ORDER BY geom_refs ASC
) As m(the_geom, geom_refs);

-- _OBS_GetPointsByGeometry

-- check that all census tracts intersecting with the geometry are returned
-- order them to ensure that the same values are returned
SELECT
  array_agg(geom_refs) = Array['36047025700','36047028501','36047038900','36047039100','36047042300','36047042500','36047042700','36047044900','36047045300','36047048500','36047048900','36047049100','36047049300','36047050500','36047050700'] As _OBS_GetPointsByGeometry_around_cartodb
FROM (
  SELECT *
  FROM cdb_observatory._OBS_GetPointsByGeometry(
    -- around CartoDB's Brooklyn office
    ST_MakeEnvelope(-73.9452409744,40.6988851644,
                    -73.9280319214,40.7101254524,
                    4326),
    'us.census.tiger.census_tract')
  ORDER BY geom_refs ASC
) As m(the_geom, geom_refs);

-- Null Island area
SELECT
  array_length(array_agg(geom_refs), 1) IS NULL As _OBS_GetPointsByGeometry_around_null_island
FROM (
  SELECT *
  FROM cdb_observatory._OBS_GetPointsByGeometry(
    -- around null island
    ST_MakeEnvelope(-0.1400756836,-0.2114863362,
                     0.1455688477, 0.2059932086,
                    4326),
    'us.census.tiger.census_tract')
  ORDER BY geom_refs ASC
) As m(the_geom, geom_refs);


-- OBS_GetPointsByGeometry

-- check that all census tracts intersecting with the geometry are returned
-- order them to ensure that the same values are returned
SELECT
  array_agg(geom_refs) = Array['36047025700','36047028501','36047038900','36047039100','36047042300','36047042500','36047042700','36047044900','36047045300','36047048500','36047048900','36047049100','36047049300','36047050500','36047050700'] As OBS_GetPointsByGeometry_around_cartodb
FROM (
  SELECT *
  FROM cdb_observatory.OBS_GetBoundariesByGeometry(
    -- around CartoDB's Brooklyn office
    ST_MakeEnvelope(-73.9452409744,40.6988851644,
                    -73.9280319214,40.7101254524,
                    4326),
    'us.census.tiger.census_tract')
  ORDER BY geom_refs ASC
) As m(the_geom, geom_refs);

SELECT
  array_agg(geom_refs) = Array['36047025700','36047028501','36047038900','36047039100','36047042300','36047042500','36047042700','36047044900','36047045300','36047048500','36047048900','36047049100','36047049300','36047050500','36047050700'] As OBS_GetPointsByGeometry_around_cartodb_2014
FROM (
  SELECT *
  FROM cdb_observatory.OBS_GetBoundariesByGeometry(
    -- around CartoDB's Brooklyn office
    ST_MakeEnvelope(-73.9452409744,40.6988851644,
                    -73.9280319214,40.7101254524,
                    4326),
    'us.census.tiger.census_tract',
    '2015')
  ORDER BY geom_refs ASC
) As m(the_geom, geom_refs);

-- Null Island area
SELECT
  array_length(array_agg(geom_refs), 1) IS NULL As OBS_GetPointsByGeometry_around_null_island
FROM (
  SELECT *
  FROM cdb_observatory.OBS_GetBoundariesByGeometry(
    -- around null island
    ST_MakeEnvelope(-0.1400756836,-0.2114863362,
                     0.1455688477, 0.2059932086,
                    4326),
    'us.census.tiger.census_tract')
  ORDER BY geom_refs ASC
) As m(the_geom, geom_refs);

-- OBS_GetPointsByPointAndRadius

-- check that all census tracts intersecting with the geometry are returned
-- order them to ensure that the same values are returned
SELECT
  array_agg(geom_refs) = Array['36047038900','36047039100','36047042500','36047042700','36047045300','36047048500','36047048900','36047049100','36047049300'] As OBS_GetPointsByPointAndRadius_around_cartodb
FROM (
  SELECT *
  FROM cdb_observatory.OBS_GetPointsByPointAndRadius(
    -- around CartoDB's Brooklyn office
    cdb_observatory._testpoint(),
    500,
    'us.census.tiger.census_tract')
  ORDER BY geom_refs ASC
) As m(the_geom, geom_refs);

SELECT
  array_agg(geom_refs) = Array['36047038900','36047039100','36047042500','36047042700','36047045300','36047048500','36047048900','36047049100','36047049300'] As OBS_GetPointsByPointAndRadius_around_cartodb_2014
FROM (
  SELECT *
  FROM cdb_observatory.OBS_GetPointsByPointAndRadius(
    -- around CartoDB's Brooklyn office
    cdb_observatory._testpoint(),
    500,
    'us.census.tiger.census_tract',
    '2015')
  ORDER BY geom_refs ASC
) As m(the_geom, geom_refs);

-- Null Island area
SELECT
  array_length(array_agg(geom_refs), 1) IS NULL As OBS_GetPointsByPointAndRadius_around_null_island
FROM (
  SELECT *
  FROM cdb_observatory.OBS_GetPointsByPointAndRadius(
    -- around null island
    ST_SetSRID(ST_Point(0, 0), 4326),
    500,
    'us.census.tiger.census_tract')
  ORDER BY geom_refs ASC
) As m(the_geom, geom_refs);

\i test/fixtures/drop_fixtures.sql
