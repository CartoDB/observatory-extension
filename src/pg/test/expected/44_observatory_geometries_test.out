\pset format unaligned
\set ECHO none
obs_getboundary_cartodb_census_tract
t
(1 row)
obs_getboundary_cartodb_county
t
(1 row)
obs_getboundary_non_existent_boundary_id
t
(1 row)
obs_getboundary_null_island_census_tract
t
(1 row)
obs_getboundary_year_census_tract
t
(1 row)
obs_getboundary_unlisted_year
t
(1 row)
obs_getboundaryid_cartodb_census_tract
t
(1 row)
obs_getboundaryid_cartodb_census_tract_with_year
t
(1 row)
obs_getboundaryid_cartodb_county_with_year
t
(1 row)
obs_getboundaryid_null_island
t
(1 row)
obs_getboundarybyid_cartodb_county
t
(1 row)
obs_getboundarybyid_compared_with_obs_getboundary
t
(1 row)
obs_getboundarybyid_boundary_id_mismatch_geom_id
t
(1 row)
obs_getboundarybyid_boundary_id_mismatch_geom_id
t
(1 row)
_obs_getboundariesbygeometry_tracts_around_cartodb
t
(1 row)
_obs_getboundariesbygeometry_tracts_around_null_island
t
(1 row)
obs_getboundariesbygeometry_tracts_around_cartodb
t
(1 row)
obs_getboundariesbygeometry_tracts_around_null_island
t
(1 row)
obs_getboundariesbypointandradius_around_cartodb
t
(1 row)
obs_getboundariesbypointandradius_around_null_island
t
(1 row)
_obs_getpointsbygeometry_around_cartodb
t
(1 row)
_obs_getpointsbygeometry_around_null_island
t
(1 row)
obs_getpointsbygeometry_around_cartodb
t
(1 row)
obs_getpointsbygeometry_around_cartodb_2014
t
(1 row)
obs_getpointsbygeometry_around_null_island
t
(1 row)
obs_getpointsbypointandradius_around_cartodb
t
(1 row)
obs_getpointsbypointandradius_around_cartodb_2014
t
(1 row)
obs_getpointsbypointandradius_around_null_island
t
(1 row)
